src/lib/storage/index/group_key/composite_group_key_index.cpp

```diff
@@ -34,27 +36,42 @@ CompositeGroupKeyIndex::CompositeGroupKeyIndex(const std::vector<std::shared_ptr
   // cast and check columns
   _indexed_columns.reserve(indexed_columns.size());
   for (const auto& column : indexed_columns) {
-    auto dict_column = std::dynamic_pointer_cast<const BaseDeprecatedDictionaryColumn>(column);
-    DebugAssert(static_cast<bool>(dict_column), "CompositeGroupKeyIndex only works with DictionaryColumns");
+    auto dict_column = std::dynamic_pointer_cast<const BaseDictionaryColumn>(column);
+    DebugAssert(static_cast<bool>(dict_column), "CompositeGroupKeyIndex only works with dictionary columns.");
+    DebugAssert(is_fixed_size_byte_aligned(dict_column->compressed_vector_type()),
+                "CompositeGroupKeyIndex only works with fixed-size byte-aligned compressed attribute vectors.");
     _indexed_columns.emplace_back(dict_column);
   }
 
   // retrieve amount of memory consumed by each concatenated key
-  CompositeKeyLength bytes_per_key = std::accumulate(
-      _indexed_columns.begin(), _indexed_columns.end(), 0u,
-      [](auto key_length, const auto& column) { return key_length + column->attribute_vector()->width(); });
+  auto bytes_per_key = std::accumulate(_indexed_columns.begin(), _indexed_columns.end(), CompositeKeyLength{0u},
+                                       [](auto key_length, const auto& column) {
+                                         return key_length + byte_width_for_fsba_type(column->compressed_vector_type());
```

Comment:
Why can't I ask the column for its byte width? That seems more intuitive to me than getting its type and asking some other function what size that type is.

Also, that would allow us to avoid that abbreviation.

Reply:
The column does not know it’s column width because it doesn’t always have one. It might also have a bit-packed attribute vector. I thought about adding it to the attribute vector aka compressed vector instead, but it would still require a cast.

Hierarchy:
```
BaseCompressedVector <|-- FixedSizeByteAlignedVector<uintx_t>
                     <|-- SimdBp128Vector
```

My idea:
```
BaseCompressedVector <|-- BaseFixedSizeByteAlignedVector <|-- FixedSizeByteAlignedVector<uintx_t>
                          + byte_width() : size_t
                     <|-- SimdBp128Vector
```

But then I‘d have to cast the vector after having checked it’s type, so that’s even more work.

Reply:
Ok, let's keep it your way, but let's get rid of the abbreviation. Since it is only used rarely, it shouldn't be a problem if the method's name gets ridiculously long.

[ASS](https://gist.github.com/klaaspieter/12cd68f54bb71a3940eae5cdd4ea1764) ;)